whylog/log_reader/__init__.py

```diff
@@ -47,25 +47,37 @@ def _save_clues_in_normal_dict(cls, collector):
             (parser_name, list(clues_iter)) for parser_name, clues_iter in collector.items()
         )
 
+    def _constraints_verification(self, clues):
+        """
+        provides constraints verification basing on rules from investigation_plan
+        and collected clues
+        """
+        causes = []
+        # effect_clues are accessible from self._investigation_plan._effect_clues
+        for rule in self._investigation_plan._suspected_rules:
+            clues_lists = []
+            for cause in rule._causes:
```

Comment:
Add generator i Rule class which returns all causes parsers names and use it here.